zarr/core.py

```diff
@@ -1980,9 +2008,29 @@ def _chunk_getitems(self, lchunk_coords, lchunk_selection, out, lout_selection,
                 for ckey in ckeys
                 if ckey in self.chunk_store
             }
+        elif (
+            self._partial_decompress
+            and not self._compressor
+            and not fields
+            and self.dtype != object
+            and hasattr(self.chunk_store, "get_partial_values")
+            and self.chunk_store.supports_efficient_get_partial_values()
+        ):
+            partial_read_decode = True
+            cdatas = {
+                ckey: UncompressedPartialReadBufferV3(
+                    ckey, self.chunk_store, itemsize=self.itemsize
+                )
+                for ckey in ckeys
+                if ckey in self.chunk_store
+            }
         else:
             partial_read_decode = False
-            cdatas = self.chunk_store.getitems(ckeys, on_error="omit")
+            if not hasattr(self.chunk_store, "getitems"):
```

Comment:
So implicitly get_partial works on multiple keys and *might* do them concurrently, but getitems always is concurrent and so preferred when there are ranges. What about just one of the keys having a range (well, typically it's the end ones, right?) and that's enough to put us on a different code path.

Reply:
fsspec's is the only one implemented, no? "By design" the method allows for concurrency.